nextcord/state.py

```diff
@@ -1086,17 +1086,31 @@ async def delete_application_command(
         if self.application_id is None:
             raise NotImplementedError("Could not get the current application id")
 
-        if guild_id:
-            await self.http.delete_guild_command(
-                self.application_id, guild_id, command.command_ids[guild_id]
+        try:
+            if guild_id:
+                await self.http.delete_guild_command(
+                    self.application_id, guild_id, command.command_ids[guild_id]
+                )
+            else:
+                await self.http.delete_global_command(
+                    self.application_id, command.command_ids[guild_id]
+                )
```

Comment:
What specific other exceptions is this block catching?

Also, maybe the KeyError and other possible errors should be added to the docstrings (Raises section) if they can occur.

Reply:
The last except block is catching any other possible exceptions, it is mirroring the catch-all except block that is present in register_application_commands.

And that is a good idea, although the KeyError is the only specific exception I have come across.

Reply:
Looks good, it seems like it logs HTTP exceptions and makes the error more specific